### ISSUE TITLE
`key_ref` parameter ignored in `vault_pki_secret_backend_intermediate_cert_request` resource creation

### ISSUE THREAD

**Issue by sestegra** [hashicorp/terraform-provider-vault#1944]
### Terraform Version

Terraform v1.5.3 on linux_amd64
Vault v1.14.0 (13a649f860186dffe3f3a4459814d87191efc321), built 2023-06-19T11:40:23Z

### Affected Resource(s)
- vault_pki_secret_backend_root_cert
- vault_pki_secret_backend_intermediate_cert_request

### Terraform Configuration Files
```hcl
resource "vault_mount" "pki" {
  path                  = "pki"
  type                  = "pki"
  description           = "PKI engine hosting root CA"
  max_lease_ttl_seconds = 157680000
}

resource "vault_mount" "pki_int" {
  path                  = "pki_int"
  type                  = "pki"
  description           = "PKI engine hosting intermediate CA"
  max_lease_ttl_seconds = 157680000
}

resource "vault_pki_secret_backend_key" "default_root_key" {
  backend  = vault_mount.pki.path
  type     = "internal"
  key_type = "ec"
  key_bits = 256
  key_name = "my_root_default_key"
}

resource "vault_pki_secret_backend_key" "root_key" {
  backend    = vault_mount.pki.path
  type       = "internal"
  key_type   = "ec"
  key_bits   = 256
  key_name   = "my_root_key"
  depends_on = [vault_pki_secret_backend_key.default_root_key]
}

resource "vault_pki_secret_backend_key" "default_int_key" {
  backend  = vault_mount.pki_int.path
  type     = "internal"
  key_type = "ec"
  key_bits = 256
  key_name = "my_int_default_key"
}

resource "vault_pki_secret_backend_key" "int_key" {
  backend    = vault_mount.pki_int.path
  type       = "internal"
  key_type   = "ec"
  key_bits   = 256
  key_name   = "my_int_key"
  depends_on = [vault_pki_secret_backend_key.default_int_key]
}

resource "vault_pki_secret_backend_intermediate_cert_request" "root_csr" {
  backend      = vault_mount.pki.path
  type         = "existing"
  organization = "Example"
  common_name  = "Example Root CA"
  key_ref      = vault_pki_secret_backend_key.root_key.key_id
}

resource "vault_pki_secret_backend_intermediate_cert_request" "int_csr" {
  backend      = vault_mount.pki_int.path
  type         = "existing"
  organization = "Example"
  common_name  = "Example Intermediate CA"
  key_ref      = vault_pki_secret_backend_key.int_key.key_id
}

```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
The generated CSR should use `my_int_key`.

### Actual Behavior
The generated CSR is created with the default private key.

Then, the default values for key_type, key_bits are incorrects when `existing` is used.
See logs below.
```
      + key_type              = "rsa"
      + key_bits              = 2048
```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

```
  ...
  # vault_pki_secret_backend_intermediate_cert_request.csr will be created
  + resource "vault_pki_secret_backend_intermediate_cert_request" "csr" {
      + add_basic_constraints = false
      + backend               = "pki_int"
      + common_name           = "Example Intermediate CA"
      + csr                   = (known after apply)
      + format                = "pem"
      + id                    = (known after apply)
      + key_bits              = 2048
      + key_id                = (known after apply)
      + key_name              = (known after apply)
      + key_ref               = (known after apply)
      + key_type              = "rsa"
      + organization          = "Example"
      + private_key           = (sensitive value)
      + private_key_format    = "der"
      + private_key_type      = (known after apply)
      + type                  = "existing"
    }
   ...
```
3. `terraform show`
```
# vault_mount.pki:
resource "vault_mount" "pki" {
    accessor                     = "pki_42da2bb6"
    audit_non_hmac_request_keys  = []
    audit_non_hmac_response_keys = []
    default_lease_ttl_seconds    = 0
    description                  = "PKI engine hosting root CA"
    external_entropy_access      = false
    id                           = "pki"
    local                        = false
    max_lease_ttl_seconds        = 157680000
    path                         = "pki"
    seal_wrap                    = false
    type                         = "pki"
}

# vault_mount.pki_int:
resource "vault_mount" "pki_int" {
    accessor                     = "pki_4729c2a7"
    audit_non_hmac_request_keys  = []
    audit_non_hmac_response_keys = []
    default_lease_ttl_seconds    = 0
    description                  = "PKI engine hosting intermediate CA"
    external_entropy_access      = false
    id                           = "pki_int"
    local                        = false
    max_lease_ttl_seconds        = 157680000
    path                         = "pki_int"
    seal_wrap                    = false
    type                         = "pki"
}

# vault_pki_secret_backend_intermediate_cert_request.int_csr:
resource "vault_pki_secret_backend_intermediate_cert_request" "int_csr" {
    add_basic_constraints = false
    backend               = "pki_int"
    common_name           = "Example Intermediate CA"
    csr                   = <<-EOT
        -----BEGIN CERTIFICATE REQUEST-----
        MIHvMIGWAgEAMDQxEDAOBgNVBAoTB0V4YW1wbGUxIDAeBgNVBAMTF0V4YW1wbGUg
        SW50ZXJtZWRpYXRlIENBMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEoVjk+dkT
        qVx1UdMI+IJhOjrutbRSSkihvtPofeK6NmJ3mk9xORvAib8xdTMXplbRE4+5zCwu
        +AXc2C/iNJYOaqAAMAoGCCqGSM49BAMCA0gAMEUCIQCnJc8PVC6StHJscrK6CUwq
        vOz1jVBjdmy2RlF2aqymqwIgctvq5Wi4H4af86qgRvsUvi9bCiRbtTiPutdsLgrL
        jR8=
        -----END CERTIFICATE REQUEST-----
    EOT
    format                = "pem"
    id                    = "pki_int/intermediate/generate/existing"
    key_bits              = 2048
    key_id                = "78d84ea9-246e-fca4-2508-902f128cc05e"
    key_ref               = "a7ee9e91-1e7f-a1cc-ef4f-87060f26bd8e"
    key_type              = "rsa"
    organization          = "Example"
    private_key_format    = "der"
    type                  = "existing"
}

# vault_pki_secret_backend_intermediate_cert_request.root_csr:
resource "vault_pki_secret_backend_intermediate_cert_request" "root_csr" {
    add_basic_constraints = false
    backend               = "pki"
    common_name           = "Example Root CA"
    csr                   = <<-EOT
        -----BEGIN CERTIFICATE REQUEST-----
        MIHnMIGOAgEAMCwxEDAOBgNVBAoTB0V4YW1wbGUxGDAWBgNVBAMTD0V4YW1wbGUg
        Um9vdCBDQTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABBnFAghoJuxXysF/3ndL
        iIL8EwIILaljbdH3QMN82MAJrsXUoL0R6zEMuVxlUsijJGKDPrqyyGXctPnjoJ1Z
        qQWgADAKBggqhkjOPQQDAgNIADBFAiEA94Xb8x8FukVnY2HvCMr6TdyJU3fdZRc7
        D8fUE7Jt28gCIBJuJLuYJ7dYYOSu/hSPxlEvMGTD1C6JAaCuBsekvKF+
        -----END CERTIFICATE REQUEST-----
    EOT
    format                = "pem"
    id                    = "pki/intermediate/generate/existing"
    key_bits              = 2048
    key_id                = "b51a70fd-71ed-4b37-03d3-6468ca5eb366"
    key_ref               = "01d75558-4296-84cc-3c85-5a790feecdca"
    key_type              = "rsa"
    organization          = "Example"
    private_key_format    = "der"
    type                  = "existing"
}

# vault_pki_secret_backend_key.default_int_key:
resource "vault_pki_secret_backend_key" "default_int_key" {
    backend  = "pki_int"
    id       = "pki_int/key/78d84ea9-246e-fca4-2508-902f128cc05e"
    key_bits = 256
    key_id   = "78d84ea9-246e-fca4-2508-902f128cc05e"
    key_name = "my_int_default_key"
    key_type = "ec"
    type     = "internal"
}

# vault_pki_secret_backend_key.default_root_key:
resource "vault_pki_secret_backend_key" "default_root_key" {
    backend  = "pki"
    id       = "pki/key/b51a70fd-71ed-4b37-03d3-6468ca5eb366"
    key_bits = 256
    key_id   = "b51a70fd-71ed-4b37-03d3-6468ca5eb366"
    key_name = "my_root_default_key"
    key_type = "ec"
    type     = "internal"
}

# vault_pki_secret_backend_key.int_key:
resource "vault_pki_secret_backend_key" "int_key" {
    backend  = "pki_int"
    id       = "pki_int/key/a7ee9e91-1e7f-a1cc-ef4f-87060f26bd8e"
    key_bits = 256
    key_id   = "a7ee9e91-1e7f-a1cc-ef4f-87060f26bd8e"
    key_name = "my_int_key"
    key_type = "ec"
    type     = "internal"
}

# vault_pki_secret_backend_key.root_key:
resource "vault_pki_secret_backend_key" "root_key" {
    backend  = "pki"
    id       = "pki/key/01d75558-4296-84cc-3c85-5a790feecdca"
    key_bits = 256
    key_id   = "01d75558-4296-84cc-3c85-5a790feecdca"
    key_name = "my_root_key"
    key_type = "ec"
    type     = "internal"
}
```

### Important Factoids
N/A

### References
Fixed on PR #1945 are related to this issue, so then issue #1944

**Comment by sestegra** [comment]
After fixing #1934 using PR #1945, the following error message is thrown when `terraform apply`

```
vault_pki_secret_backend_intermediate_cert_request.csr: Creating...
╷
│ Error: error creating intermediate cert request for PKI secret backend "pki_int": Error making API request.
│
│ URL: PUT http://127.0.0.1:8200/v1/pki_int/intermediate/generate/existing
│ Code: 400. Errors:
│
│ * invalid parameter for the kms/existing path parameter, key_type nor key_bits arguments can be set in this mode
│
│   with vault_pki_secret_backend_intermediate_cert_request.csr,
│   on test.tf line 16, in resource "vault_pki_secret_backend_intermediate_cert_request" "csr":
│   16: resource "vault_pki_secret_backend_intermediate_cert_request" "csr" {
│
╵
```

**Comment by sestegra** [comment]
Values for `key_type`, `key_bits` and `key_name` should be read from the resulting `existing` key.

**Comment by sestegra** [comment]
@vinay-gopalan
This issue has been resolved through the implementation of #1973.

**Comment by fairclothjm** [comment]
Thanks @sestegra !
